website/docs/concepts/reading.mdx

```diff
@@ -213,10 +219,9 @@ For example, we can use `ref.read` to increment a counter when a user click on a
 <CodeBlock>{trimSnippet(read)}</CodeBlock>
 
 :::note
-Using `ref.read` should be avoided as much as possible.
+Using `ref.read` should be avoided as much as possible in the widget tree (except for in callbacks)
```

Comment:
That's incorrect. This mention isn't specific to widget trees. "read" should indeed be avoided everywhere IMO

Reply:
Shouldn't callbacks avoid ref.watch? And isn't using a Reader recommended for passing into a StateNotifier for example?

Reply:
You don't have to rely on callbacks to begin with.  
You can do:

```dart
final notifier = ref.watch(provider.notifier);

return Button(onPressed: notifier.increment);
```


Same goes for StateNotifier & co. If you somehow can refactor your code to use watch instead of read, do it.